src/pkgconf/__init__.py

```diff
@@ -32,6 +32,9 @@
 
 
 def _get_system_executable() -> Optional[pathlib.Path]:
+    if os.getenv('PKGCONF_PYPI_EMBEDDED_ONLY') == '1':
+        return None
```

Comment:
I think we should probably return `False` when disabled, then we can also skip the warning in `get_executable`.

nitpick: Use `os.environ` instead, also I generally prefer enabling on non-empty values.

```suggestion
    if os.environ.get('PKGCONF_PYPI_EMBEDDED_ONLY'):
        return None
```